File: server/datasets/python_source-code/Multiple-G3/renamed_37_dsu.py
class DSU_1:
    def __init__(self, n_1):
        self.p_1 = list(range(n_1))
        self.rk_1 = [0] * n_1

    def find_1(self, x_1):
        if self.p_1[x_1] != x_1:
            self.p_1[x_1] = self.find_1(self.p_1[x_1])
        return self.p_1[x_1]

    def union_1(self, x_1_a, x_1_b):
        ra_1 = self.find_1(x_1_a)
        rb_1 = self.find_1(x_1_b)
        if ra_1 != rb_1:
            if self.rk_1[ra_1] < self.rk_1[rb_1]:
                ra_1, rb_1 = rb_1, ra_1
            self.p_1[rb_1] = ra_1
            if self.rk_1[ra_1] == self.rk_1[rb_1]:
                self.rk_1[ra_1] += 1
            return True
        return False

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_g3/renamed_37_dsu.java
public class DSU_1 {
    public static void main(String[] args) {
        DSU_1 dsu = new DSU_1(10);
        dsu.u_1(1, 2);
    }
    int[] p_1;
    int[] r_1;
    public DSU_1(int tot_1) {
        p_1 = new int[tot_1];
        r_1 = new int[tot_1];
        for (int idx = 0; idx < tot_1; idx++) {
            p_1[idx] = idx;
            r_1[idx] = 0;
        }
    }
    public int f_1(int x_1) {
        if (p_1[x_1] != x_1) {
            p_1[x_1] = f_1(p_1[x_1]);
        }
        return p_1[x_1];
    }
    public void u_1(int a_1, int b_1) {
        int ra_1 = f_1(a_1);
        int rb_1 = f_1(b_1);
        if (ra_1 != rb_1) {
            if (r_1[ra_1] < r_1[rb_1]) {
                p_1[ra_1] = rb_1;
            } else if (r_1[ra_1] > r_1[rb_1]) {
                p_1[rb_1] = ra_1;
            } else {
                p_1[rb_1] = ra_1;
                r_1[ra_1]++;
            }
        }
    }
}